src/trade_ui.cpp

```diff
@@ -69,6 +70,8 @@ std::string trade_preset::get_denial( const item_location &loc ) const
         npc const &np = *_trader.as_npc();
         if( !np.wants_to_buy( *loc, price, market_price ) ) {
             return string_format( _( "%s does not want to buy this" ), np.get_name() );
+        } else if( loc->has_flag( flag_FILTHY ) ) {
```

Comment:
Put this in with the rest of the refusal logic in `npc::wants_to_buy()`

Reply:
Oh yeah, thats a better spot for it

Reply:
you could change the return value of `can_trade()` to `ret_val<bool>` which can store a string for additional context